ci/pipeline.yml

```diff
@@ -0,0 +1,258 @@
+
+jobs:
+  - name: build-release-ci-images
+    plan:
+    - get: ci-images-git-repo
+      trigger: true
+    - put: release-ci-image
+      params:
+        build: ci-images-git-repo/ci/images/release-ci-image
+
+  - name: build
+    serial: true
+    public: true
+    plan:
+    - get: release-ci-image
+    - get: git-repo
+      trigger: true
+    - do:
+      - task: build-project
+        privileged: true
+        timeout: 1h30m
+        image: release-ci-image
+        file: git-repo/ci/tasks/build-project.yml
+    - put: artifactory-repo
+      params: &artifactory-params
+        repo: libs-snapshot-local
+        folder: distribution-repository
+        build_uri: "https://scs.ci.springapps.io/teams/main/pipelines/${BUILD_PIPELINE_NAME}/jobs/${BUILD_JOB_NAME}/builds/${BUILD_NAME}"
+        build_number: "${BUILD_PIPELINE_NAME}-${BUILD_JOB_NAME}-${BUILD_NAME}"
+        disable_checksum_uploads: true
+    on_failure:
+      *slack-failure-notification
+
+  - name: stage-milestone
+    serial: true
+    plan:
+      - get: release-ci-image
+      - get: git-repo
+        trigger: false
+      - task: stage
+        image: release-ci-image
+        file: git-repo/ci/tasks/stage.yml
+        params:
+          RELEASE_TYPE: M
+      - put: artifactory-repo
+        params:
+          <<: *artifactory-params
+          repo: libs-staging-local
+      - put: git-repo
+        params:
+          repository: stage-git-repo
+  - name: stage-rc
+    serial: true
+    plan:
+      - get: release-ci-image
+      - get: git-repo
+        trigger: false
+      - task: stage
+        image: release-ci-image
+        file: git-repo/ci/tasks/stage.yml
+        params:
+          RELEASE_TYPE: RC
+      - put: artifactory-repo
+        params:
+          <<: *artifactory-params
+          repo: libs-staging-local
+      - put: git-repo
+        params:
+          repository: stage-git-repo
+  - name: stage-release
+    serial: true
+    plan:
+      - get: release-ci-image
+      - get: git-repo
+        trigger: false
+      - task: stage
+        image: release-ci-image
+        file: git-repo/ci/tasks/stage.yml
+        params:
+          RELEASE_TYPE: RELEASE
+      - put: artifactory-repo
+        params:
+          <<: *artifactory-params
+          repo: libs-staging-local
+      - put: git-repo
+        params:
+          repository: stage-git-repo
+  - name: promote-milestone
+    serial: true
+    plan:
+      - get: release-ci-image
+      - get: git-repo
+        trigger: false
+      - get: artifactory-repo
+        trigger: false
+        passed: [stage-milestone]
+        params:
+          save_build_info: true
+      - task: promote
+        image: release-ci-image
+        file: git-repo/ci/tasks/promote.yml
+        params:
+          RELEASE_TYPE: M
+          ARTIFACTORY_SERVER: ((artifactory-server))
+          ARTIFACTORY_USERNAME: ((artifactory-username))
+          ARTIFACTORY_PASSWORD: ((artifactory-password))
```

Comment:
According to the concourse docs, this is still a valid pattern. https://concourse-ci.org/task-step.html#task-step-params

Reply:
It's totally valid in that it still works, it's just a lot less convenient. To `fly execute` a task that uses params, we have to provide them all as env vars, e.g.

```
ARTIFACTORY_SERVER=... ARTIFACTORY_USERNAME=... ARTIFACTORY_PASSWORD=... fly execute -c task.yml -l non-secret-config.yml
```

With var templating that goes away because the secrets are templated in from CredHub, so then it's just 

```
fly execute -c task.yml -l non-secret-config.yml
```

It's a small thing, but makes life easier, and I think the only reason we're using params in tasks any more is because not everyone's aware of this feature yet - it's pretty new. I'd like to have new CI tasks take advantage of CredHub so we can converge on having all our CI secrets managed there.

Reply:
Oh I see!